src/cnd/swaps_payload.ts

```diff
@@ -79,3 +80,227 @@ export type HalightLightningBitcoinHerc20EthereumErc20RequestBody = CoreRequestB
   HalightLightningBitcoinRequestParams,
   Herc20EthereumErc20RequestParams
 >;
+
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties: Properties;
+  entities: Array<LedgerState | LedgerParameters>;
+  actions: SwapAction[];
+  /**
+   * links for this swap, contains a self reference
+   */
+  links: Link[];
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties: Properties;
+  entities: Array<LedgerState | LedgerParameters | Identities>;
+  actions: SwapAction[];
+  /**
+   * links for this swap, contains a self reference
+   */
+  links: Link[];
+}
+
+/**
+ * The properties of a swap
+ */
+export interface Properties {
+  // id: string; // No id! The id should not be manipulated by the client, navigation should be done using the `links` property.
+  /**
+   * The status this swap is currently in.
+   */
+  status: SwapStatus;
+  /**
+   * The role in which you are participating in this swap.
+   */
+  role: "Alice" | "Bob";
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters of a given ledger
+ */
+export interface LedgerParameters extends EmbeddedRepresentationSubEntity {
+  /**
+   * The relation of these ledger parameters to the parent object (*SwapProperties).
+   */
+  rel: ["alpha" | "beta"];
+  /**
+   * Human readable title.
+   */
+  title: "Parameters of the Alpha Ledger" | "Parameters of the Beta Ledger";
+  /**
+   * Class of this sub-entity to facilitate parsing.
+   */
+  class: ["parameters"];
+  properties: {
+    /**
+     * Protocol: "han", "herc20" or "halight"
+     */
+    protocol: string;
```

Comment:
You could make the protocol a class, then the properties are fixed.

Reply:
You could also merge the `LedgerState` into this then because we can guarantee a set of fixed fields per protocol, i.e. `transactions.deploy` for example.

Reply:
The protocols would have to be:

- `han-ethereum`
- `han-bitcoin`
- `herc20`
- `halight`

but that is how I like to think about them anyway :)

Also gets rid of the `ledger` field.

Reply:
You could then expand the `asset` field into:

- `quantity`
- `contract_address` (only for `herc20`)

Reply:
what happens when we support halight for litecoin?

Reply:
Is that the proposal you are making?

```
{
  class: ["parameters","han-ethereum"|"han-bitcoin"|...];
  properties: NativeAsset|Erc20Asset;
...
}

export interface NativeAsset {
	name: "bitcoin"|"ether";
	quantity: string;
}

export interface Erc20Asset {
	contract_address: "0x...";
	quantity: string;
}
```

Reply:
Yes with the change that I wouldn't do `NativeAsset` I think :)

The way I like to think about `han` is that it is an abstract protocol that can be instantiated for ethereum and bitcoin but it cannot appear by itself, hence I would always talk about `han-ethereum` or `han-bitcoin`.

Reply:
> Yes with the change that I wouldn't do `NativeAsset` I think :)
> 
> The way I like to think about `han` is that it is an abstract protocol that can be instantiated for ethereum and bitcoin but it cannot appear by itself, hence I would always talk about `han-ethereum` or `han-bitcoin`.

Can you please write some code, I don't understand what you mean. How do you express bitcoin asset if you dont do `NativeAsset`?

Reply:
@yosriady @thomaseizinger can you please both clarify your position on this subject?

Reply:
You only need the quantity field if you have `han-bitcoin` :)

Reply:
> > The protocols would have to be:
> > 
> >     * `han-ethereum`
> > 
> >     * `han-bitcoin`
> > 
> >     * `herc20`
> > 
> >     * `halight`
> > 
> > 
> > but that is how I like to think about them anyway :)
> > 
> > Also gets rid of the `ledger` field.
> 
> what happens when we support halight for litecoin?
> 

Name it `halight-bitcoin` then if you want :)